### DISCUSSION TITLE
Reusing previous answer online

### DISCUSSION THREAD

**Discussion by Venustiano** [rubenarslan/formr.org#545] [Q&A] [answered]
I would like to reuse the previous answer as in the screenshot below

![image](https://github.com/rubenarslan/formr.org/assets/39260912/c9147b51-a724-473e-a861-e0ab2690c60d)

However, when I try to access the variable I get `NA`. My code is as follows

![image](https://github.com/rubenarslan/formr.org/assets/39260912/65b517dc-b4b5-41bd-822f-fbd32e7be84e)

Do you have an idea how can I use the value of the `Toerbelsysvast` variable?

Thanks in advance

**Comment by dfeistauer** [comment]
try a submit between both variables.

**Reply by Venustiano** [reply]
Adding a `submit` will move to the next page but I still need to be in the same page and update the note.

**Reply by Venustiano** [reply]
I also don't understand why I can use the variables in the `showif` column but not in the `label`.

**Reply by rubenarslan** [reply]
The label is generated with rmarkdown (requires a pageload), but the showif is evaluated both with JS (works on the same page). You can also generate the label with JS, but you'd have to write that code yourself.

**Reply by Venustiano** [reply]
I see, 

Thank you for your reply. Could you point me to an example running javascript in a formr google/excel sheet?

Thanks again

**Reply by Venustiano** [reply]
I found a solution

Based on [this answer](https://groups.google.com/g/formr/c/wT08FAPebJc/m/ZYWKUq0UAQAJ),  I implemented the following javascript function

```
window.onload = function () {
    const dnumber = document.querySelector('[name="Toerbelsysvast"]');
    
    display = document.querySelector('#test_display_number');
    
    dnumber.addEventListener("change", (event) => {
        gnumber = event.target.value
        display.textContent = gnumber;
    })
};
```
and in the `label` column

```
Test display number: <span id="test_display_number"> </span>
```
However, I still have a question. This code is working only for `Runs`  buy not for `Test survey`.

![image](https://github.com/rubenarslan/formr.org/assets/39260912/a98c8700-e70f-4645-912a-6a0acb7e133e)

Is it possible to include javascript code into `Test survey`s?

**Comment by rubenarslan** [comment, marked as answer]
Yes, you can include any Javascript in the survey label, so just make the label of Toerbelsysvaat contain the below:

```
<script>
window.onload = function () {
    const dnumber = document.querySelector('[name="Toerbelsysvast"]');
    
    display = document.querySelector('#test_display_number');
    
    dnumber.addEventListener("change", (event) => {
        gnumber = event.target.value
        display.textContent = gnumber;
    })
};
</script>
```

**Reply by Venustiano** [reply]
Thank you for your quick reply
